packages/playback-core/src/renditions.ts

```diff
@@ -0,0 +1,103 @@
+import Hls from './hls';
+import type { HlsInterface } from './hls';
+import type { MediaTracks } from './types';
+import type { TrackEvent } from 'media-tracks';
+
+export function setupRenditions(customMediaEl: MediaTracks, hls: HlsInterface) {
+  if (!('videoTracks' in customMediaEl)) return;
+
+  // Create a map to save the unique id's we create for each level and rendition.
+  // hls.js uses the levels array index primarily but we'll use the id to have a
+  // 1 to 1 relation from rendition to level.
+  const levelIdMap = new WeakMap();
+
+  hls.on(Hls.Events.MANIFEST_PARSED, function (event, data) {
+    removeAllVideoTracks();
+
+    const videoTrack = customMediaEl.addVideoTrack('main');
+    videoTrack.selected = true;
+
+    for (const [id, level] of data.levels.entries()) {
+      const videoRendition = videoTrack.addRendition(
+        level.url[0],
+        level.width,
+        level.height,
+        level.videoCodec,
```

Comment:
this should be the composed mimecodec str, correct?

Reply:
I'd rather do this in a next iteration. it's not required for the renditions UI selector afaik